app/models/template_kind.rb

```diff
@@ -20,7 +20,8 @@ def self.default_template_labels
       "script" => N_("Script template"),
       "user_data" => N_("User data template"),
       "ZTP" => N_("ZTP PXE template"),
-      "POAP" => N_("POAP PXE template")
+      "POAP" => N_("POAP PXE template"),
+      "cloud-init" => N_("cloud-init template")
```

Comment:
One UX thing - now we end up with user_data and cloud-init. We need to somehow explain to users the difference between the two. Maybe different naming would help as well? User data was always pretty tough to understand, maybe "seed data". Well maybe not.

Do we need to separate templates for the same format? I mean, eventually you will end up with two same templates in cloud-init format but one as user_data and the other as cloud-init. Is it possible to merge the two so the same template is seeded into VMWare and then acquired via cloud-init client?

Reply:
Yeah, I was thinking the same. I think we should use userdata for the cloud-init templates and find a better way for the `open-vm-tools` customization. Let me think of something...

Reply:
In the worst case we need to add some helper text to users explaining what is what.

Reply:
Would it be too bad to call it open-vm-tools template? And keep userdata for cloudinit only?

Reply:
If we call this `open-vm-tools` template, we'd have to change how the vmware compute ressource loads the customization template. It cannot use the userdata template anymore.
I'm not against this approach, but it would further delay this PR as someome (probably me) has to rewrite the code. I'd do that in a separate PR, though. We'd then change this PR to only render the userdata template in the unattended controller and drop the cloud-init template kind for good.

@dgoetz also might have some ideas here.

Reply:
I have a customer at the moment starting with Katello (indeed with Linux as central managed os at all) who is going to use the plugin Timo created for this before now trying to get this into core. His planned (and at the moment only possible) workflow is using VMware Templates, set a static IP with _user_data_ and then do all the other configuration via _cloud_init_, which is not possible with core only. From this experience a big +1 for this going into core.
Another option instead of _cloud_init_ would be _finish_ but _finish_ templates get disabled when enabling _user_data_ for the VMware Template. _finish_ would be more easy to understand for most users especially those new to automated provisioning as it is simply shell scripting. cloud_init requires typically to learn something new, in addition to all other Foreman/Katello things.
So having two very similar templates I also fear confuses people, so at least naming and documentation should be in a way to help avoid confusion. Changing how VMware compute resource is working (or simply renaming the template) I also think will confuse many because it is perhaps the one I see most and so I fear many guides which will not be updated are out there.
So _user_data_ would be send to the Cloud/Virtualization solution and _cloud_init_ would be for querying by on the system installed cloud_init. Would this be the case for all compute resources? If yes, naming it specific like _open-vm-tools_ seems wrong to me. Any naming suggestions based on the communication direction? Or would it be enough to have this as help text, in the header comment in the template and documentation in the manual? I have no real solution but at least we should try to make it as simple for users as possible, many are already confused by all the options and templates Foreman provides.

Reply:
> So user_data would be send to the Cloud/Virtualization solution and cloud_init would be for querying by on the system installed cloud_init.

Yes.

> If yes, naming it specific like open-vm-tools seems wrong to me

I concur.

> would it be enough to have this as help text, in the header comment in the template

I like that idea.

Reply:
I agree that we should better explain template kinds to users and we already do! This was added in 1.20 I think (extra help line / bubble text). Here it is:

```ruby
    @@template_description ||= {
      "PXELinux" => N_("Used when PXELinux loader is set, loads pxelinux.0 which loads content generated by this template."),
      "PXEGrub" => N_("Used when Grub2 loader is set, loads grub/bootx64.efi which loads content generated by this template."),
      "PXEGrub2" => N_("Used when PXELinux loader is set, loads grub/grubx64.efi which loads content generated by this template."),
      "iPXE" => N_("Used in iPXE environments."),
      "provision" => N_("Template for OS installer, for example kickstart, preseed or jumpstart. Depends on the operating system."),
      "finish" => N_("Post-install script for preseed-based or cloud instance. Connection is made via SSH, credentials or key must exist and inventory IP address must match. Only used when 'user data' is not set."),
      "script" => N_("An arbitrary script, must be manually downloaded using wget/curl."),
      "user_data" => N_("Template with seed data for virtual or cloud instances when 'user data' flag is set, typically cloud-init or ignition format."),
      "ZTP" => N_("Provisioning Junos devices (Junos 12.2+)."),
      "POAP" => N_("Provisioning for switches running NX-OS.")
    }
```

I guess we are doing better job now, it's obvious that user_data is explained as "seed data". Finish script is also well covered. I think we are fine here.